packages/lib-utils/src/k8s/hooks/useK8sWatchResources.test.ts

```diff
@@ -0,0 +1,160 @@
+    (useSelector as jest.Mock)
+      .mockReturnValueOnce(ImmutableMap<string, K8sModelCommon>()) // Models have not loaded so allK8sModels is empty
+      .mockReturnValueOnce(ImmutableMap<string, unknown>());
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourcesMock);
+
+    const { result } = renderHook(() => useK8sWatchResources(watchedResourcesMock));
+    const { application } = result.current;
+    const { data, loaded, loadError } = application;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(false);
+    expect(loadError).toBeUndefined();
+  });
+
+  test('should return specific error if the model for the watched resource does not exist', () => {
+    (useModelsLoaded as jest.Mock).mockReturnValue(true);
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourcesMock);
+    (useSelector as jest.Mock)
+      .mockReturnValueOnce(ImmutableMap<string, K8sModelCommon>()) // Models have loaded but do not contain "application" model
+      .mockReturnValueOnce(ImmutableMap<string, unknown>());
+
+    const { result } = renderHook(() => useK8sWatchResources(watchedResourcesMock));
+    const { application } = result.current;
+    const { data, loaded, loadError } = application;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(true);
+    expect((loadError as NoModelError).message).toEqual('Model does not exist');
+  });
+
+  test('should return data for the watched resource', () => {
+    const mockWatchData: WatchData = {
+      id: 'appstudio.redhat.com~v1alpha1~Application---{"ns":"test-ns","name":"test"}',
+      action: jest.fn(),
+    };
+    const payload = {
+      'appstudio.redhat.com~v1alpha1~Application---{"ns":"test-ns","name":"test"}': {
+        data: resourceDataMock,
+        loaded: true,
+        loadError: '',
+      },
+    };
+    const resourceK8s: ImmutableMap<string, unknown> = ImmutableMap(fromJS(payload));
+
+    jest.mock('./usePrevious', () => ({
+      usePrevious: jest.fn(() => undefined),
+    }));
+    (useModelsLoaded as jest.Mock).mockReturnValue(true);
+    (useDeepCompareMemoize as jest.Mock).mockReturnValue(watchedResourcesMock);
+    const useSelectorSpy = (useSelector as jest.Mock)
+      .mockReturnValueOnce(ImmutableMap<string, K8sModelCommon>(allModelsMock)) // Mock models
+      .mockReturnValueOnce(ImmutableMap<string, unknown>(resourceK8s));
+    (getWatchData as jest.Mock).mockReturnValue(mockWatchData);
+    (getReduxData as jest.Mock).mockReturnValue(resourceDataMock);
+
+    const { result } = renderHook(() => useK8sWatchResources(watchedResourcesMock));
+    const { application } = result.current;
+    const { data, loaded, loadError } = application;
+    expect(loaded).toEqual(true);
+    expect(loadError).toEqual('');
+    expect((data as K8sResourceCommon).metadata?.creationTimestamp).toEqual('2022-04-29T13:41:21Z');
+    expect(useSelectorSpy).toHaveBeenCalledTimes(2);
```

Comment:
Not sure how helpful it is to assert that `useSelector` is called twice here versus a simple `toHaveBeenCalled()`. If we go to add another `useSelector` call in the future this test will fail.